[BugFix] Update Statement fail when SET a column using DEFAULT key word(#15181)

srlch opened the pull request:
## What type of PR is this：
- [x] BugFix
- [ ] Feature
- [ ] Enhancement
- [ ] Refactor
- [ ] UT
- [ ] Doc
- [ ] Tool

## Which issues of this PR fixes ：
<!--
Usage: `Fixes #<issue number>`, or `Fixes (paste link of issue)`.
_If PR is about `failing-tests or flakes`, please post the related issues/tests in a comment and do not use `Fixes`_*
-->
Fixes #15181

## Problem Summary(Required) ：
Problem:
When we Update a column which is defined the default value, if we use set = default in update statement, we will get unknow error because the DefaultValueExpr is not handled properly.

Solution:
Add the some code to cast the DefaultValueExpr to corresponding Literal type.

## Checklist:

- [ ] I have added test cases for my bug fix or my new feature
- [ ] This pr will affect users' behaviors
- [ ] This pr needs user documentation (for new or modified features or behaviors)
  - [ ] I have added documentation for my new feature or new function

## Bugfix cherry-pick branch check:
- [x] I have checked the version labels which the pr will be auto backported to target branch
  - [x] 2.5
  - [x] 2.4
  - [x] 2.3
  - [x] 2.2

wanpengfei-git commented:
#### [FE PR Coverage Check]

:heart_eyes: **pass** : 5 / 5 (100.00%)




#### file detail

|   |path|covered_line|new_line|coverage|not_covered_line_detail|
|----|----|----|----|----|----|
|:large_blue_circle:|com/starrocks/sql/ast/ColumnAssignment.java|2|2|100.00%|[]|
|:large_blue_circle:|com/starrocks/sql/analyzer/UpdateAnalyzer.java|3|3|100.00%|[]|

decster commented on fe/fe-core/src/main/java/com/starrocks/sql/analyzer/UpdateAnalyzer.java:
```diff
@@ -81,6 +83,11 @@ public static void analyze(UpdateStmt updateStmt, ConnectContext session) {
                 }
 
+                if (assign.getExpr() instanceof DefaultValueExpr) {
+                    assign.setExpr(new StringLiteral(col.calculatedDefaultValue()));
+                    assign.setExpr(TypeManager.addCastExpr(assign.getExpr(), col.getType()));
```
`TypeManager.addCastExpr(new StringLiteral(col.calculatedDefaultValue()), col.getType())`

srlch committed:
[BugFix] Update Statement fail when SET a column using DEFAULT key word(#15181)

Problem:
When we Update a column which is defined the default value, if we use set = default in update statement, we will get
unknow error because the DefaultValueExpr is not handled properly.

Solution:
Add the some code to cast the DefaultValueExpr to corresponding Literal type.

wanpengfei-git commented:
#### [FE PR Coverage Check]

:heart_eyes: **pass** : 4 / 4 (100.00%)




#### file detail

|   |path|covered_line|new_line|coverage|not_covered_line_detail|
|----|----|----|----|----|----|
|:large_blue_circle:|com/starrocks/sql/ast/ColumnAssignment.java|2|2|100.00%|[]|
|:large_blue_circle:|com/starrocks/sql/analyzer/UpdateAnalyzer.java|2|2|100.00%|[]|

wanpengfei-git commented:
run starrocks_admit_test

wanpengfei-git commented:
@Mergifyio backport branch-2.5

wanpengfei-git commented:
@Mergifyio backport branch-2.4

wanpengfei-git commented:
@Mergifyio backport branch-2.3

wanpengfei-git commented:
@Mergifyio backport branch-2.2
